enhancements/cluster-scope-secret-volumes/shared-resource-validation.md

```diff
@@ -0,0 +1,407 @@
+
+Depending on the outcome to the open questions, new tests will verify
+- any deletions or marking as invalid previously created Shared* objects that are now invalid (most likely unit test from an automation perspective)
+- Manual smoke tests per our QE team's Behave based smoke test scenarios (they periodically import these into Polarian for consistency with the rest of OCP QE) will be added to create Shared* objects in 4.10, then upgrade OCP, then verify results
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+N / A
+
+#### Tech Preview -> GA
+
+Adding just this feature will not result in this feature moving from Tech Preview as introduced in 4.10 to GA.
+
+We also need the storage team's work and SCC/Pod Policy and formalizing the approach for unprivileged users 
+creating objects with use of the Shared Resource CSI Driver.
+
+Nor is it the gating factor on the API moving past the v1alpha1 level introduced in 4.10,
+though it is possible per the Open Questions that we add a validity field, but event then, we may keep the API at 
+v1alpha1 with that, and see how much via metrics we catch 4.10 users using "openshift-*", and remove it later.  
+
+Otherwise, ideally, we want a period of usage in the field
+and feedback from users before making the call for API version promotion from v1alpha to v1beta or v1.
+
+
+
+#### Removing a deprecated feature
+
+N / A
+
+### Upgrade / Downgrade Strategy
+
+The validation admission controls discussed here are inherently stateless across calls to it, so the belief is 
+upgrades to new versions should roll out the associated Deployments.  Similarly, changes in the Webhook's configuration
+should get applied by existing OCP CVO/CSO processing.  The only API for engaging the webhooks is the k8s webhook APIs themselves.
+
+On the "openshift-*" naming restriction, where 4.10 Shared Resources may no longer be valid, explicit setting of the 
+2 new boolean fields on their Shared Resource instances will re-insert the Reference data on any existing Pods that 
+were using the user created Shared Resources that starts with "openshift-*".
+
+The existing or lack there of the ValidatingAdmissionWebhook should not impact a 4.10 version of the current Controller/CSI Driver.
+Either the kubelet calls it or it does not call it during upgrades.  Either it gets watch events from Kubernetes or it does not
+during upgrades.  The enhancement here in no way depends on any elements of how the kubelet and a CSI driver operator that 
+do not exist in 4.10, when Shared Resources were introduced.
+
+The CSI Driver does make a SAR request, as well as access the contents of the associated Secret or ConfigMap, during its processing.
+But if the API Server is not up to honor those requests, then any requests from the kubelet to mount our volumes will be denied.
+But that is no different than the level of service provided in 4.10.
+
+### Version Skew Strategy
```

Comment:
Until the CSI driver is behind I don't think it's necessary to add convoluted conditions to validate existing Shared Resources, there can't be any version skew.

Reply:
Per my agreement at https://github.com/openshift/enhancements/pull/1056#discussion_r823895693 I will clean this up too